### PR TITLE
Add leader filter in hot region scheduler

### PR DESCRIPTION
**Pull request by Yisaer**
### What problem does this PR solve?
During `filterDstStores` in hot region scheduler, it didn't consider the placement rule enabled in `transferLeader` case.

### What is changed and how it works?
This requests add `PlacementLeaderSafeguard` filter in the `filterDstStores`

### CODE REVIEW

File: server/schedule/filter/filters.go [hunk 1]
```diff
@@ -77,6 +77,9 @@ func Target(opt opt.Options, store *core.StoreInfo, filters []Filter) bool {
 	storeAddress := store.GetAddress()
 	storeID := fmt.Sprintf("%d", store.GetID())
 	for _, filter := range filters {
+		if filter == nil {
```
Comment (rleungx) [inline server/schedule/filter/filters.go:80]: When will this case happen?

Comment (Yisaer) [reply server/schedule/filter/filters.go:80]: `NewPlacementLeaderSafeguard` will return nil if placementRule is not enabled.

Comment (rleungx) [reply server/schedule/filter/filters.go:80]: But it will check if `leaderFilter != nil` before appending.

Comment (HunDunDM) [inline server/schedule/filter/filters.go:80]: When will `filter` be `nil`? I did not find this.

Comment (Yisaer) [reply server/schedule/filter/filters.go:80]: see `NewPlacementLeaderSafeguard`, it would return nil if placementRules is not enabled.

**Comment by rleungx** [comment]
How about adding a unit test?

**Comment by Yisaer** [comment]
> How about adding a unit test?

updated.

**Pull request merged**